File: vortexdbg-api/src/main/java/com/vortexdbg/unix/struct/ITimerVal64.java
package com.vortexdbg.unix.struct;

import com.vortexdbg.pointer.VortexdbgStructure;
import com.sun.jna.Pointer;

import java.util.Arrays;
import java.util.List;

public class ITimerVal64 extends VortexdbgStructure {

    public ITimerVal64(Pointer p) {
        super(p);
    }

    public TimeVal64 it_interval; /* timer interval */
    public TimeVal64 it_value; /* current value */

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("it_interval", "it_value");
    }
}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/unix/struct/ITimerVal64.kt
package com.vortexdbg.unix.struct

import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

class ITimerVal64(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var it_interval: TimeVal64? = null /* timer interval */
    @JvmField
    var it_value: TimeVal64? = null /* current value */

    override fun getFieldOrder(): List<String> {
        return listOf("it_interval", "it_value")
    }
}
